src/Tasks/DownloadFile.cs

```diff
@@ -69,6 +70,11 @@ public void Cancel()
         }
 
         public override bool Execute()
+        {
+            return ExecuteAsync().GetAwaiter().GetResult();
```

Comment:
This is the (unforunate) sync wrapper around the async code, because, to my knowledge, there is no async entry point into a build Task.  So this is the only place that can't use it.